### PR TITLE
Add monitor for cri-containerd/containerd

### LINKED ISSUE #419: Add containerd/cri-containerd monitor.
In our current kube-up.sh integration, we don't have a [`kube-docker-monitor.service`](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/node.yaml#L42) like health monitor for cri-containerd/containerd.

We should add one.

### PR DESCRIPTION
**Pull request by miaoyq**
fixes #419 

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

### CODE REVIEW

File: test/health-monitor.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      pkill containerd
+      # Wait for a while, as we don't want to kill it again before it is really up.
+      sleep 60
+    else
+      sleep "${SLEEP_SECOMDS}"
+    fi
+  done
+}
+
+function cri_containerd_mmonitoring {
```
Comment (mikebrow) [inline test/health-monitor.sh]: function name typo

File: test/health-monitor.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  while [ 1 ]; do
+    if ! timeout 60 crictl --runtime-endpoint /var/run/cri-containerd.sock ps > /dev/null; then
```
Comment (mikebrow) [inline test/health-monitor.sh]: crictl should already have it's own timeout I think

File: test/health-monitor.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      echo "cri-containerd failed!"
+      pkill cri-containerd
+      # Wait for a while, as we don't want to kill it again before it is really up.
+      sleep 60
```
Comment (mikebrow) [inline test/health-monitor.sh]: long wait here should be able to use a smaller wait and check if it's running again?

Comment (mikebrow) [reply test/health-monitor.sh]: maybe try to reuse some of setup apis over here: https://github.com/kubernetes-incubator/cri-containerd/blob/master/hack/test-utils.sh#L54-L64

File: test/health-monitor.sh (hunk 1 above)
Comment (miaoyq) [reply test/health-monitor.sh]: Thanks, will do.

File: test/health-monitor.sh (hunk 2 above)
Comment (miaoyq) [reply test/health-monitor.sh]: Seems have no timeout in `crictl`, see:
https://github.com/kubernetes-incubator/cri-tools/blob/master/cmd/crictl/container.go#L548-L655

Comment (miaoyq) [reply test/health-monitor.sh]: `crictl` only have timeout of connecting to server (default: 10s), but we still need to the timeout that  if the connecting to server is ok, but the request of `crictl ps` is blocked.

File: test/health-monitor.sh (hunk 3 above)
Comment (miaoyq) [reply test/health-monitor.sh]: It's better, will use `readiness_check` here :-)

Comment (miaoyq) [reply test/health-monitor.sh]: Done.

File: test/health-monitor.sh (hunk 1 above)
Comment (miaoyq) [reply test/health-monitor.sh]: Done.

**Comment by Random-Liu** [comment]
I'm not sure whether this works. How do you ship the script to the node?

**Comment by miaoyq** [comment]
Does not seem to work :-P
I see the `configure.sh` is get online
https://github.com/kubernetes-incubator/cri-containerd/blob/c89e55515016b380ef779f37865cd4ca05b7c5bf/test/e2e/node.yaml#L23

the same to `kubernetes`:
https://github.com/kubernetes/kubernetes/blob/13dd275859b3cffeffc40ad849f847a27d971919/cluster/gce/gci/configure.sh#L217-L221

how can I upload the script to `http://metadata.google.internal`?

@Random-Liu  Could you give some guidance? Thanks!

**Comment by Random-Liu** [comment]
@miaoyq I'm also working on this, will send out a PR soon.

**Comment by miaoyq** [comment]
@Random-Liu OK, I will close this and study your PR deeply after sending out, try to get how it works. :-)

**Pull request closed without merging**